Makes Lesser Form into one ability & unit tests it

Jacquerel opened the pull request:
## About The Pull Request

Fixes #73491
Every time I have used this ability lately it's been fucked. 
It would vanish from my actions at arbitrary moments, and also sometimes transform me into a horrible monkey-man thing instead of a monkey. This is a shame because being able to become a monkey can be pretty fun, even if it makes you very vulnerable to being butchered.

Refactoring it into being one action instead of two actions which add and remove each other fixes the part where the action just disappears. It reliably sticks between transformations now, regardless of whether or not they were voluntary.

I also noticed that when I was turning into a monkey it wasn't dropping the changeling "fake clothes" outfit pieces I had on as a human, leading to a really fucked up looking monkey. I fixed this by adding `force = TRUE` in the drop to ground proc in the check for if the equipment you have is still valid after your species changes. I don't _think_ this has any side effects but I never do and then someone finds some.
For good measure I also made all of the changeling equipment abilities which don't work if you are a monkey detect if you become a monkey and retract themselves.

I also noticed that for a long time Last Resort has been trying and failing to give you Lesser Form (well, Human Form rather) as a Headcrab, so I fixed that and now you actually get the ability.

Finally I did a _little_ bit of housekeeping in general on the changeling actions, mostly balloon alerts. I think these definitely need more attention than I gave them though. I left a lot of the `to_chat`s in place because many of them give information you want to be a little sticky, or refer back to in order to double check what you just did.

I also added a unit test which flips back and forth a few times to ensure the ability still works.
This required adding an "instant" flag to the monkeyize/humanize procs to skip the timers, and idenitified a couple of weird issues.
First point: Humanising a monkey would remove the monkey mutation and then call humanise again, which would not skip itself because it still regarded you as being a monkey. I changed the order of operations here slightly so that it will early return.
Second point: Calling `domutcheck` on `human/consistent` would runtime because we skip the bit which sets up any mutations in their DNA. This is a part of changeling transformation, so I just made it return instantly.

## Why It's Good For The Game

You can use this ability again without getting stuck permanently as a monkey, or it just deleting itself from your list of abilities for no reason.
Turning into a monkey with fake outfit pieces on won't turn you into an abomination.

## Changelog

:cl:
refactor: Changeling's Lesser Form is now one ability instead of two which keep swapping, which should consistently turn you back and forth without deleting itself from your action bar.
fix: Hatching from an egg left by a Last Resort headcrab should correctly grant you Lesser Form in addition to your other abilities.
fix: Turning into a monkey while using the Changeling space suit won't leave you as a monkey with a weird inflated head.
qol: Using lesser form as a monkey with only one stored DNA profile will skip asking which profile you want and will simply transform you immediately into the only option.
/:cl:

Jacquerel committed:
Refactor lesser form into one ability
Make sure changeling gear is dropped on transformation

Jacquerel committed:
My old friend balloon alert
Headcrab now correctly grants lesser form

Jacquerel committed:
Oops wrong one

ZephyrTFA commented on code/modules/antagonists/changeling/powers/mutations.dm:
```diff
@@ -196,24 +222,24 @@
-			playsound(A, 'sound/machines/airlock_alien_prying.ogg', 100, TRUE)
-			if(!do_after(user, 100, target = A))
+			playsound(opening, 'sound/machines/airlock_alien_prying.ogg', 100, TRUE)
+			if(!do_after(user, 100, target = opening))
```
```suggestion
			if(!do_after(user, 10 SECONDS, target = opening))
```

Jacquerel committed:
Update code/modules/antagonists/changeling/powers/mutations.dm

Co-authored-by: Zephyr <12817816+ZephyrTFA@users.noreply.github.com>

san7890 commented on code/modules/antagonists/changeling/powers/lesserform.dm:
```diff
@@ -5,49 +5,62 @@
+/// Stop being a monkey
+/datum/action/changeling/lesserform/proc/unmonkey(mob/living/carbon/human/user)
+	if(user.movement_type & VENTCRAWLING)
+		user.balloon_alert(user, "exit pipes!")
```
leaving the pipes is already quite a visual/intuitive thing so i don't think this balloon alert makes sense/is useful from a UX standpoint, probably best to do without it

Jacquerel replied:
This is an instruction (you can't do this because you are in some pipes) rather than a notification but the fact that you might have confused it for something else tells me that I went too far in trying to keep it brief.

san7890 replied:
oh i misunderstood that entire proc. new balloon alert definitely better though

san7890 commented on code/modules/antagonists/changeling/powers/mutations.dm:
```diff
@@ -196,24 +222,24 @@
+		user.visible_message(span_warning("[user] forces the airlock to open with [user.p_their()] [src]!"), span_warning("We force the [opening] to open."), \
 		span_hear("You hear a metal screeching sound."))
-		A.open(2)
+		opening.open(2)
```
```suggestion
		opening.open(forced = 2)
```

"2" makes no sense anyways but let's at least be nice

san7890 replied:
actually the whole concept of "2" has been bugging me for a while now so i'm just going to rework this whole stupid thing

Jacquerel committed:
balloon alert matches the one from shriek abilities

san7890 commented:
> As Melbert suggested I had a look at making a unit test for this but ran into some snags because I haven't made one before.
The chief snag is that monkeyize and humanize both start timers to give time to play an animation before completing, and I don't really know a way around that which wouldn't invalidate the test (because I want to make sure that pressing the actual button works)

Make a unit-test only subtype (that lives in the unit test file) that calls overridden versions of the animation proc to make it not sleep. See `door_access.dm` unit test for an example (but I defined the special doors used in actual code because i'm fucked up in the mind)

Jacquerel committed:
Adds a lesser form unit test
Change order of operations so that humanize doesn't call itself, little bit scary
Skip mutcheck on dummy humans

san7890 commented on code/modules/mob/living/carbon/human/dummy.dm:
```diff
@@ -117,6 +117,9 @@ INITIALIZE_IMMEDIATE(/mob/living/carbon/human/dummy)
 	if(is_creating)
 		fully_replace_character_name(real_name, "John Doe")
 
+/mob/living/carbon/human/consistent/domutcheck()
+	return // We skipped adding any mutations so this runtimes
```
This won't screw with using the consistent dummy in other unit tests, right? (i.e.: this should have been implemented at the start due to certain assumptions about the DNA with the dummy, but wasn't)

Jacquerel replied:
In my investigation it seems that this proc is _mostly_ called manually, and not by any of the unit tests or anything that was being unit tested.
I could not find anywhere which was setting some actual mutations and _then_ calling it, and if not doing that it would have runtimed. So as far as I could ascertain it shouldn't have side effects on other tests.
The alternative to this is actually giving the dummy a predefined list of dormant mutations in its DNA, but it's never needed it before.
